src/_common/LanguagePicker.js

```diff
@@ -2,6 +2,16 @@ import React from 'react';
 import { SelectGroup } from '../_common';
 import languages from '../_constants/languages';
 
-export default () => (
-    <SelectGroup options={languages} value="EN" />
+const LanguagePicker = ({onChange, current}) => (
+    <SelectGroup options={languages} value={current} onChange={onChange} />
 );
+
+LanguagePicker.propTypes = {
+    current: React.PropTypes.oneOf(languages.map(ln => ln.value)),
+};
+
+LanguagePicker.defaultProps = {
+    current: 'EN',
```

Comment:
'selected' seems the proper name for the LanguagePicker property.
On the other hand the persisted value can be 'currentLanguage' or just 'language'.